### PR TITLE
Mail cleanup

### PR DESCRIPTION
**Pull request by marcelhollerbach**

### CODE REVIEW

**Comment by marcelhollerbach** [comment]
This introduces a bit strange calls into the main, I will refactor the main in another pr, dont wanted to do this also in this one, since this would have been very messy

**Comment by LeanderK** [comment]
what about disabling stuff? Is the docker-config ok without modifications?

**Comment by marcelhollerbach** [comment]
havent changed anything on that. should run like before

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/mail/MailReceiver.java [hunk 1]
```diff
@@ -14,25 +9,36 @@
...
-        this.auth = auth;
+    public MailReceiver(Protocol protocol, String user, String password, String host, int port) {
+        this.protocol = protocol;
+        this.user = user;
+        this.password = password;
+        this.host = host;
+        this.port = port;
+
+        props = new Properties();
+        props.setProperty("mail.debug", true+"");
```
Comment (kelunik) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/mail/MailReceiver.java:40]: How about `"true"`?

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java [hunk 2]
```diff
@@ -196,15 +188,16 @@ private static void boot(DatabaseManager databaseManager, List<Platform> platfor
         DatabaseMaintainer maintainer = new DatabaseMaintainer(databaseManager.getContext(), cleanupInterval);
         maintainer.start();
 
-        MailFetcher mailFetcher = getMailFetcher(mailDisabled, sendMail);
-        MailSender mailSender = getMailSender(mailDisabled, sendMail);
+        MailFetcher mailFetcher = getMailFetcher(mailDisabled, getConfig().mail.moneyReceiver);
```
Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: doesn't that produce an NPE with the docker config?

Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: mhmm other words for that is that the config is incomplete. Shouldnt we find some other solution ?

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/mail/MailReceiver.java (hunk 1 above)
Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/mail/MailReceiver.java:40]: nah I like to be typo-protected.

File: src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java (hunk 2 above)
Comment (LeanderK) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: well, we have to find some kind of solution ;)

Comment (marcelhollerbach) [inline src/main/java/edu/kit/ipd/crowdcontrol/objectservice/Main.java]: solution: DELIVERED.

**Pull request merged**